Neos.Flow/Classes/Mvc/Routing/RouteConfiguration.php

```diff
@@ -0,0 +1,120 @@
+<?php
+namespace Neos\Flow\Mvc\Routing;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Mvc\Exception\InvalidRouteSetupException;
+
+/**
+ * The default router configuration
+ *
+ * @Flow\Scope("singleton")
+ * @api
+ */
+class RouteConfiguration
+{
+    /**
+     * @Flow\Inject
+     * @var ConfigurationManager
+     */
+    protected $configurationManager;
+
+    /**
+     * Array of routes to match against
+     *
+     * @var Routes
+     */
+    protected $routes;
+
+    /**
+     * true if route object have been created, otherwise false
+     *
+     * @var boolean
+     */
+    protected $routesCreated = false;
+
+    /**
+     * Returns a list of configured routes
+     *
+     * @return Routes
+     */
+    public function getRoutes()
+    {
+        $this->createRoutesFromConfiguration();
```

Comment:
IMO this could be changed to
```suggestion
        if (!$this->routesCreated) {
            $this->createRoutesFromConfiguration();
        }
```
and then..

Reply:
👍